packages/vega4-extension/src/index.ts

```diff
@@ -73,37 +73,60 @@ class RenderedVega extends Widget implements IRenderMime.IRenderer {
   /**
    * Render Vega/Vega-Lite into this widget's node.
    */
-  renderModel(model: IRenderMime.IMimeModel): Promise<void> {
-    const data = model.data[this._mimeType] as JSONObject;
-    const metadata = model.metadata[this._mimeType] as { embed_options?: EmbedOptions };
+  async renderModel(model: IRenderMime.IMimeModel): Promise<void> {
+    const spec = model.data[this._mimeType] as JSONObject;
+    const metadata = model.metadata[this._mimeType] as { embed_options?: VegaModuleType.EmbedOptions };
     const embedOptions = metadata && metadata.embed_options ? metadata.embed_options : {};
-    const mode: Mode = this._mimeType === VEGA_MIME_TYPE ? 'vega' : 'vega-lite';
-    return this._resolver.resolveUrl('').then((path: string) => {
-      return this._resolver.getDownloadUrl(path).then(baseURL => {
-        const options: EmbedOptions = {
-          actions: true,
-          defaultStyle: true,
-          ...embedOptions,
-          mode,
-          loader: {
-            baseURL,
-            http: { credentials: 'same-origin' }
-          }
-        };
-        const el = document.createElement('div');
-        this.node.innerHTML = '';  // clear the output before attaching a chart
-        this.node.appendChild(el);
-        return vegaEmbed(el, data, options).then(result => {
-          // Add png representation of vega chart to output
-          if (!model.data['image/png']) {
-            return result.view.toImageURL('png').then(imageData => {
-              const data = { ...model.data, 'image/png': imageData.split(',')[1] };
-              model.setData({ data });
-            });
-          }
-          return void 0;
-        });
-      });
+    const mode: VegaModuleType.Mode = this._mimeType === VEGA_MIME_TYPE ? 'vega' : 'vega-lite';
+
+    const vega: typeof VegaModuleType = await this._ensureVega();
+    const path = await this._resolver.resolveUrl('');
+    const baseURL = await this._resolver.getDownloadUrl(path);
+
+    const el = document.createElement('div');
+
+    // clear the output before attaching a chart
+    this.node.innerHTML = '';
+    this.node.appendChild(el);
+
+    const result = await vega.default(el, spec, {
+      actions: true,
+      defaultStyle: true,
+      ...embedOptions,
+      mode,
+      loader: {
+        baseURL,
+        http: { credentials: 'same-origin' }
+      }
+    });
+
+    if (model.data['image/png']) {
+      return;
+    }
+
+    // Add png representation of vega chart to output
+    const imageURL = await result.view.toImageURL('png');
+    model.setData({
+      data: { ...model.data, 'image/png': imageURL.split(',')[1] }
+    });
+  }
+
+  /**
+   * Lazy-load the vega-embed library.
+   */
+  private _ensureVega(): Promise<typeof VegaModuleType> {
```

Comment:
I tried to unwrap this with `await`s, but webpack is picky about how it should work, so leaving for now.